catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/utils/TypeUtils.java

```diff
@@ -0,0 +1,248 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.datastrato.gravitino.catalog.lakehouse.paimon.utils;
+
+import com.datastrato.gravitino.rel.types.Type;
+import com.datastrato.gravitino.rel.types.Types;
+import java.util.Arrays;
+import org.apache.paimon.types.ArrayType;
+import org.apache.paimon.types.BigIntType;
+import org.apache.paimon.types.BinaryType;
+import org.apache.paimon.types.BooleanType;
+import org.apache.paimon.types.DataType;
+import org.apache.paimon.types.DataTypeDefaultVisitor;
+import org.apache.paimon.types.DataTypes;
+import org.apache.paimon.types.DateType;
+import org.apache.paimon.types.DecimalType;
+import org.apache.paimon.types.DoubleType;
+import org.apache.paimon.types.FloatType;
+import org.apache.paimon.types.IntType;
+import org.apache.paimon.types.LocalZonedTimestampType;
+import org.apache.paimon.types.MapType;
+import org.apache.paimon.types.MultisetType;
+import org.apache.paimon.types.RowType;
+import org.apache.paimon.types.SmallIntType;
+import org.apache.paimon.types.TimeType;
+import org.apache.paimon.types.TimestampType;
+import org.apache.paimon.types.TinyIntType;
+import org.apache.paimon.types.VarCharType;
+
+/** Utilities of {@link Type} to support type conversion. */
+public class TypeUtils {
+
+  private TypeUtils() {}
+
+  /**
+   * Convert Paimon {@link DataType} data type to Gravitino {@link Type} data type.
+   *
+   * @param dataType Paimon {@link DataType} data type.
+   * @return Gravitino {@link Type} data type.
+   */
+  public static Type fromPaimonType(DataType dataType) {
+    return dataType.accept(PaimonToGravitinoTypeVisitor.INSTANCE);
+  }
+
+  /**
+   * Convert Gravitino {@link Type} data type to Paimon {@link DataType} data type.
+   *
+   * @param type Gravitino {@link Type} data type.
+   * @return Paimon {@link DataType} data type.
+   */
+  public static DataType toPaimonType(Type type) {
+    return GravitinoToPaimonTypeVisitor.visit(type);
+  }
+
+  private static class PaimonToGravitinoTypeVisitor extends DataTypeDefaultVisitor<Type> {
+
+    private static final PaimonToGravitinoTypeVisitor INSTANCE = new PaimonToGravitinoTypeVisitor();
+
+    @Override
+    public Type visit(VarCharType varCharType) {
+      if (varCharType.getLength() == Integer.MAX_VALUE) {
```

Comment:
what's the reason?

Reply:
paimon treats `stringType` as `varchar(Integer.MAX_VALUE)`.

Reply:
when converting a paimon datatype to a gravitino datatype, we should distinguish between a stringType and a varcharType